app/controllers/api/v2/migrations/splits_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Api::V2::Migrations::SplitsController < AuthenticatedApiController
+  def create
+    split_upsert = SplitUpsert.new(create_params.merge(app: current_app, require_app_name_prefix: true))
+    if split_upsert.save
+      head :no_content
+    else
+      render_errors split_upsert
+    end
+  end
+
+  private
+
+  def create_params
+    # rails-sponsored workaround https://github.com/rails/rails/pull/12609
+    params.permit(:name, weighting_registry: params[:weighting_registry].try(:keys))
```

Comment:
Oh interesting. This doesn't work?:

```ruby
params.permit(:name, weighting_registry: {})
```

Reply:
```ruby
pry(main)> ActionController::Parameters.new(test: { a: 2, c: { banana: 'foo' } }, fish: 1).permit(test: {})
=> <ActionController::Parameters {"test"=><ActionController::Parameters {"a"=>2, "c"=><ActionController::Parameters {"banana"=>"foo"} permitted: true>} permitted: true>} permitted: true>

pry(main)> _.to_h
=> {"test"=>{"a"=>2, "c"=>{"banana"=>"foo"}}}

Reply:
I dunno, I copied this from the previous working controller and would be hesitant to change it unless you've got confidence it won't break anything. :) It may have been to work around a limitation of rails that no longer exists?

Reply:
As long as the `weighting_registry` is constructed of permitted scalars (i.e. string, numbers, dates, etc) it should be fine. It can be deeply nested, contain lists, etc -- the `{}` argument is meant as an escape hatch for exactly this kind of use case. Maybe it historically wasn't so permissive? I'd be curious to see what the test cases say.

Reply:
Oh I see, maybe in this case it explicitly _shouldn't_ contain any nesting, despite how the keys are arbitrary? That makes more sense. Maybe the comment could mention that.

Reply:
Doesn't quite feel like strong params is the best way to guard for that -- once you're out of "known params" territory into the land of arbitrary/dynamic params, something at the model layer could validate that the values are all numbers, etc.

Reply:
Yeah, jogging my memory, I think it’s that, but I’m general the strong params model’s job is to ensure you don’t get funny shaped data, not that the scalars you get in the correct spots are invalid. Then it’s a validation and serialization concern. So yeah, I could comment it, but I’m not convinced strongparams isn’t doing exactly what I want it to here.